### PR TITLE
Documentation: Add `ssl_context` option to `RedisCluster` documentation

### LINKED ISSUE #299: Add documentation to `RedisCluster` that we allow `ssl_context` option
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

With https://github.com/laminas/laminas-cache-storage-adapter-redis/pull/85, the `RedisCluster` implementation allows `ssl_context` options to be passed so that these are applied when connecting to clusters with SSL enabled.

Since documentation is still not available from the adapters itself, we have to add these here. (cc @froschdesign)

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| Documentation | yes

### Description

This adds the documentation of the new created `ssl_options` option flag for `RedisCluster` options.
Closes #299 
Is a new patch version feasible, this component does actually not change at all.

### CODE REVIEW

File: docs/book/v3/storage/adapter.md [hunk 1]
```diff
@@ -796,17 +796,18 @@ Capability | Value
...
+|-----------------------|---------------------------|---------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `lib_options`         | `array`                   | `[]`          | Associative array of Redis options where the array key is the options constant value (see `RedisCluster::OPT_*` [constants](https://github.com/JetBrains/phpstorm-stubs/blob/master/redis/RedisCluster.php) for details).                          |
+| `namespace_separator` | `string`                  | ":"           | A separator for the namespace and prefix.                                                                                                                                                                                                          |
```
Comment (froschdesign) [inline docs/book/v3/storage/adapter.md:802]:
```suggestion
| `namespace_separator` | `string`                  | `:`           | A separator for the namespace and prefix.                                                                                                                                                                                                          |
```

File: docs/book/v3/storage/adapter.md [hunk 2]
```diff
@@ -796,17 +796,18 @@ Capability | Value
...
+| `password`            | `string`                  | ""            | Password to authenticate with Redis server                                                                                                                                                                                                         |
```
Comment (froschdesign) [inline docs/book/v3/storage/adapter.md:803]: I suggest a comment like "(empty)" as the default value.

File: docs/book/v3/storage/adapter.md [hunk 3]
```diff
@@ -796,17 +796,18 @@ Capability | Value
...
+| `name`                | `string`                  | ""            | Name to determine configuration from [php.ini](https://github.com/phpredis/phpredis/blob/develop/cluster.markdown#loading-a-cluster-configuration-by-name) (**MUST NOT** be combined with `seeds`)                                                 |
```
Comment (froschdesign) [inline docs/book/v3/storage/adapter.md:804]: The same suggestion here.

File: docs/book/v3/storage/adapter.md [hunk 4]
```diff
@@ -796,17 +796,18 @@ Capability | Value
...
+| `read_timeout`        | `float`                   | `2.0`         | Read timeout for commands, see [PhpRedis](https://github.com/phpredis/phpredis/blob/develop/cluster.markdown#timeouts) timeouts documentation for more background.                                                                                 |
+| `persistent`          | `bool`                    | `false`       | Flag to specify whether to create a persistent connection or not                                                                                                                                                                                   |
+| `version`             | `string`                  | ""            | The Redis server version. **MUST** be specified in a [Semantic Versioning 2.0.0](https://semver.org/#semantic-versioning-200) format. This information is used to determine some features/capabilities without opening a connection to the server. |
```
Comment (froschdesign) [inline docs/book/v3/storage/adapter.md:809]: And here.

**Review by froschdesign — CHANGES_REQUESTED** [review]

**Comment by boesing** [comment]
@froschdesign I keep the suggestions here open as they should be done indeed but actually only appeared due to the fact that I reformatted all the tables to match markdown codingstyle. Contents where not changed by me and thus lets keep this focused on adding `ssl_context`.

I am still unhappy with the way how adapter configuration has to be maintained. 
I raised this concern a lot yet I am unable to provide useful input. 
The main pain points for me are:
- cross repository changes
- changes which do not reflect adapter versioning (there are multiple majors in several adapters)

I'd like to still see all the docs regarding an adapter listed in the appropriate satellites. So we can provide dedicated cache page the same way we do for https://docs.laminas.dev/mvc/
This would allow us to have every (supported) adapter listed on that page individually in dedicated versionings.
No idea how we did that for MVC but IMHO it should look like that so that we finally can provide feature documentation from within the adapter repository and do not have cross repository PRs such as these.

This said, I will merge this for now so that we have the SSL context properly documented.

**Pull request merged**